include/openenclave/edger8r/common.h

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/**
+ * @file common.h
+ *
+ * This file defines the inline functions, macros and data-structures used in
+ * oeedger8r generated code on both enclave and host side.
+ * These internals are subject to change without notice.
+ *
+ */
+#ifndef _OE_EDGER8R_COMMON_H
+#define _OE_EDGER8R_COMMON_H
+
+#include <openenclave/bits/defs.h>
+#include <openenclave/bits/result.h>
+#include <openenclave/bits/types.h>
+
+OE_EXTERNC_BEGIN
+
+/******************************************************************************/
+/********* Macros and inline functions used by generated code *****************/
+/******************************************************************************/
+
+/**
+ * Add a size value, rounding to sizeof(void*).
+ */
+OE_INLINE oe_result_t oe_add_size(size_t* total, size_t size)
+{
```

Comment:
@gupta-ak Can you double check the logic of this?
Turns out this is the only safe-math functionality I need in edger8r generated code.